### DISCUSSION TITLE
Install 4.0.1 - problème 1er connexion admin

### DISCUSSION THREAD

**Discussion by Chico0008** [JeromeDevome/GRR#234] [Q&A]
Bonjour

Je suis en train de preparer une Mise a jour de notre ancien GRR vers une nouvelle version
j'ai recuperer la toute derniere version 4.0.1 et installer sur mon apache, pas de soucis.

je lance l'url, je fait l'install, connexion BDD ok, definition du mdp pour ADMINISTRATEUR OK
une fois fini, j'ai une petite page demandant une MAJ, pourquoi pas, je fait mettre a jour, et je revient sur l'index qui me demande de m'authentifier

je met donc le compte ADMINISTATEUR, et le mdp defini pendant l'install, et ca ne passe pas :/ je reste sur la page d'authentification

je vient de voir dans la log
`[Tue Jan 31 10:13:35.479827 2023] [php:warn] [pid 62478] [client 10.0.3.123:61479] PHP Warning:  Undefined array key "login" in /var/www/grr/include/session.inc.php on line 632, referer: http://grr.viry.net/login.php?url=%2Fday.php%3Farea%3D1`

**Comment by AndrewHobson** [comment]
Bonjour,

Je suis également très intéressé par ce sujet !
Car je dois sérieusement songer à migrer notre installation sur un serveur plus récent avec une version plus récente de Grr.

Du coup, puis-je te demander depuis quelle version de Grr tu es parti pour faire cette migration ?

Merci pour ta réponse et A+

**Reply by Chico0008** [reply]
Pour le moment je suis en phase de test
notre GRR actuel est hebergé sur un vieu windows serveur avec ISS
Je compte migré sur un Ubuntu server 22 avec apache/php/mariadb

deja je veut faire une install fraiche pour tester un peu a vide et la conf
dans une 2nd phase, je testerait une migration (j'ai testé tout a l'heure une migration avec une autre version de GRR, mais la mise en page deconnait completement, les couleur s'affichaient pas)

**Reply by ynaessens** [reply]
Bonjour,
a priori la version la plus stable est la v3.5.0b.
Je prends cependant note de votre difficulté avec la v4.0.1 et cherche une solution au problème.
Les autres versions sont à considérer comme obsolètes (3.4.x et 4.0.0) ou encore en phase de maturation (3.5.1 et 4.0.1). Nous sommes évidemment preneurs de vos retours d'expérience.
Cordialement,
YN

**Reply by Chico0008** [reply]
Pas de soucis, de mon coté c'est un projet long terme, donc à la limite je peut attendre une correction de la 4.0.1, si en plus cela peut vous permettre d'avoir des retour d'experience :)

**Reply by ynaessens** [reply]
Suite à mise à jour de v4.0.0 vers v4.0.1, je constate que la première connexion de l'administrateur ne passe pas, mais la deuxième oui. Et les suivantes passent du premier coup. Vous confirmez ?

**Reply by Chico0008** [reply]
j'ai essayer plusieurs fois, aucune ne passe de mon coté, avec le Mdp defini lors de l'install.
le Mdp comprend des min/Maj/chiffre et un caractère special (un !)

erreur dans la log apache lors de la tentative de connexion
```
[Tue Jan 31 14:32:53.154855 2023] [php:warn] [pid 63866] [client 10.0.3.123:64792] PHP Warning:  Undefined array key "login" in /var/www/grr/include/session.inc.php on line 370, referer: http://grr.viry.net/login.php
[Tue Jan 31 14:32:53.155296 2023] [php:warn] [pid 63866] [client 10.0.3.123:64792] PHP Warning:  Undefined array key "login" in /var/www/grr/include/session.inc.php on line 632, referer: http://grr.viry.net/login.php
```

**Reply by ynaessens** [reply]
C'est bizarre. Pour les mots de passe cryptés, l'algorithme par défaut de password_hash done une suite de caractères commençant par `$2y$10$`

**Reply by Chico0008** [reply]
dans la base le mdp crypté commence par b45...
visiblement pas le bon type de hashage lors de la définition :/

je vient d'essayer de générer le hash de mon Mdp avec une commande php
```
php -a
echo password_hash('mon_mot_de_passe',PASSWORD_DEFAULT);
```
et mis le hash dans la base directement, mais idem, l'authentification ne passe pas, pas de message d'erreur sur la page (alors que si je met un mauvais mdp j'ai un bandeau rouge), et toujours le meme message d'erreur dans la log apache.

**Reply by ynaessens** [reply]
Bizarre !
Je viens d'ajouter un nouvel utilisateur par l'interface de GRR, le mot de passe commence toujours par `$2y$10$` et est reconnu dès la première connexion :-)
Par contre, il ne faut pas jouer avec password_hash(), et c'est là son intérêt : si vous prenez le même mot de passe et que vous le hachez deux fois, le résultat sera presque sûrement différent. La seule façon de reconnaître un mot de passe est d'utiliser password_verify()... et donc de connaître le mot de passe !
Si vous avez bien respecté les règles, je ne comprends pas d'où vient le blocage.
Pour vos tests, vous pouvez essayer avec la v3.5.1 ?

**Reply by Chico0008** [reply]
je vient de tester avec la 3.5.1 RC1
premiere connexion en admin ne passe pas, mais les suivantes sont OK
aucun erreur apache.

**Reply by ynaessens** [reply]
Merci, ça me rassure.
Il y a donc du travail à faire pour mettre la 4.0.1 d'aplomb. Je regarde dès que possible.

**Reply by ynaessens** [reply]
A priori ce commit corrige le bug : https://github.com/JeromeDevome/GRR/commit/c3e48362fbdc230a733fb80d2b8f5036745e6cc8
Merci de confirmer si j'ai vu juste.
Encore désolé pour le dérangement,
cordialement,
YN

**Reply by Chico0008** [reply]
En faisant une install from scratch c'est Ok désormais pour se loguer en administrateur sans erreur

**Reply by ynaessens** [reply]
Bonne nouvelle !
Pour la mise à niveau d'un ancien GRR, le plus simple est de partir de votre nouvelle installation, d'y restaurer une sauvegarde de l'ancien, puis de faire la mise à jour de la base de données vers la nouvelle version. En principe, ce doit être bon. Sinon, n'hésitez pas à ouvrir un nouveau sujet.
Cordialement,
YN

**Comment by JeromeDevome** [comment]
Bonjour,
après une installation neuve (ou non), le mot de passe admin était en md5. Lors de la première tentative de connexion GRR le convertit via password_hash, et la session n'aboutit pas. A la deuxième connexion le mot de passe est haché et donc la session aboutit.

Deux points pour pallier le problème : 
- Lors d'une nouvelle installation nous utilisons le password_hash (réalisé via https://github.com/JeromeDevome/GRR/commit/93f39c0c8594e2174eff52b7d3b759cd616b84da)
- Pour les installation du passé, ou quand il y aura un changement de hash faire la session avec nouveau hash (A analyser)

**Comment by pleco27** [comment]
Bonjour, j'ai le même problème de login: ADMINISTRATEUR qui ne passe pas avec le mdp initialisé

Au debut, je coyais que c'était un pb de version mysql et php: je me suis mis en conformité avec ubuntu 18.04
Mais toujouts même problème, et je ne comprend pas ce qu'il faut faire exactement ..

Une idée ?

pleco27

**Reply by ynaessens** [reply]
Bonjour,
pouvez-vous essayer les fichiers à jour de ce commit : https://github.com/JeromeDevome/GRR/tree/93f39c0c8594e2174eff52b7d3b759cd616b84da
et nous dire si votre problème est bien résolu ?
Cordialement,
YN

**Reply by pleco27** [reply]
Bonjour, je vous remercie de me répondre aussi rapidement, mais je ne comprends pas ce qu'il faut faire
Quand je clique sur le lien, c'est la page principale dédié à GRR, quand je clique sur la "branche", je trouve des numéros de version genre: GRR-3.5.0 et aussi un "master" qui parait récent. Que dois je faire à ce stade ?
Désolé, je n'ai pas l'habitude...
pleco27

**Reply by ynaessens** [reply]
Dans la page dont je donne le lien, il y a un bouton « Code », puis vous choisissez « Download zip » pour avoir l’ensemble des fichiers à jour. Vous installez comme une release et si vous voulez bien, vous nous rendez compte ;-)
Cordialement,
YN

**Reply by pleco27** [reply]
Bonjour
OK, je suis arrivé à trouver et charger le zip. La taille est sensiblement la même que la version4.0.1: Je pense que c'est normal ? Je suis en train de me demander si la version v4.0.1 n'est pas en instable, alors que mon but est de déployer GRR aupèrs d'une assoc où je suis bénévole: Est ce que je prends un risque, et est ce que je ne ferai pas mieux d'installer en production une version antérieure ? Et si oui, laquelle est la plus stable ? Si je ne  prends pas trop de risque, je veux bien tenter la release sur mon serveur de test actuel (qui est éteint quelques jour), sauf que je ne sais pas faire une release: Est ce faire une release, c'est placer tous les fichiers du fichier zip dans le répertoire /var/www/html/GRR-4.0.1/ ?
Merci encore. Et oui, je reporterai ici les tests effectués, j'aime juste comprendre ou je vais !
pleco27

**Reply by ynaessens** [reply]
Bonjour,
vous installez les fichiers comme ceux d'une version : décompressez et transférez dans le dossier /www ou /htdocs où sont vos services web.
Ensuite vous suivez les instructions du fichier INSTALL.txt, comme pour une version stable (stable release).
Il semble que la version actuelle la plus stable soit la 3.5.0b, ce qui ne veut pas dire qu'elle soit sans défauts...
Bon courage et bonne chance pour la suite !
YN

**Reply by pleco27** [reply]
Bonsoir,
Ok, j'ai suivi vos conseils, et du coup, avec cette version, je passe le blocage du login ADMINISTRATEUR qui ne passait pas avec le mdp initialisé. Maintenant c'est **OK**
Par contre, cette version n'est pas encore nommée ? Serait-ce une future 4.0.2 ?
J'hésite vraiment à la passer en prod, qu'en pensez vous ?
Merci encore
pleco27

**Reply by pleco27** [reply]
Ah j'ai la réponse: C'est marqué: "Version de développement, ne pas utiliser en production !"
Du coup, je vais allez vers la 3.5.0b, si c'est possible
pleco27

**Reply by JeromeDevome** [reply]
Bonjour,
Je viens de publier la version 4.0.2 en release candidate.

Si nous avons des retours positifs elle sera publiée rapidement en stable.

C'est grâce à vos remarques que GRR avance, merci à vous.

**Reply by pleco27** [reply]
Bonjour Jerome
Excellente nouvelle, je viens de l'installer, pas de blocage au niveau connexion, je vais la tester un peu
Mais j'ai toujours en administration le message: "Version de développement, ne pas utiliser en production !"
Je pense que c'est normal, mais du coup, pas de prod: A ton avis, en général, c'est bon pour la prod combien de temps après la publication ?
Merci
pleco27

**Reply by pleco27** [reply]
Premier test: Ajout d'un utilisateur administrateur, connexion avec cet utilisateur: pas de menu admin ?
pleco

**Comment by JeromeDevome** [comment]
Bonjour @pleco27 ,
Oui l'avertissement est "normal", il y a pas de risque majeur car nous ne touchons pas actuellement à la base de donnée sur la version de développement.
Le temps dépend des tests de qualifications, les objectifs, les remontés d'information de la part des utilisateurs.

Nous avons encore un soucis de connexion LDAP que @ynaessens va regarder avant la mise en version stable, si d'ici là nous n'avons pas découvert de problème majeur.

Pour votre test avec l'admin, je vient de corriger (que sur les prochaines création) : 5bfb70dd
Pour vous débloquer : Éditer l'utilisateur sans rien modifier et faite  "Enregistrer"

**Reply by pleco27** [reply]
Bonsoir, j'ai réalisé les modifications décrites dans fichier php avec nano, et ensuite la manip décrite sur l'utilisateur déjà créé: C'est **OK** je retrouve les droits administrateur chez l'utilisateur concerné
Merci
Du coup, est ce que la source est également mofifiée ? (dans le cas où j'installe la 4.0.2 en prod
Autre question, la source n'était en pas nommée GRR-4.0.2, mais du genre grrxxxxxxxxxxxxxxx.zip: normal ?
Dernière question; Suite au pb que je rencontrais avec 4.0.1, je me suis efforcé de me mettre en conformité avec les prérequis sur mysql et php, mais du coup mon serveur de test est sous ubuntu 18.04... Est ce que utiliser un serveur sous ubuntu 22.04 me provoquera des soucis particulier?
pleco27

**Reply by JeromeDevome** [reply]
La source sera modifiée dans la 4.0.2 stable.
Il n'y a pas de compatibilité os mais PHP et sql. A vérifier si la version linux supporte apache avec les versions php recommandées.

**Reply by pleco27** [reply]
Re
Un souci dans le menu: Administration/Type de réservation/Modification d'un type de réservation:
Impossible de choisir la couleur de fond en hexadécimal
J'ai essayé sous Firefox et Chrome, au cas où ce soit le navigateur...
Merci
pleco27

**Reply by JeromeDevome** [reply]
Bonjour,
Je ne reproduis pas le problème.
Pouvez vous me retourner le résultat de la console (Sur firefox : Clic droit sur la page, inspecter, Console)

**Reply by pleco27** [reply]
Bonjour,
Voici:
Uncaught TypeError: $.widget is undefined
    <anonymous> http://192.168.1.45/GRR4/admin/admin.php?p=admin_type_modify&id_type=1:1089

**Reply by pleco27** [reply]
L'anomalie réside dans le fait que je ne peux ni effacer ni remplacer directement la zone "Choix de la couleur de fond en hexadécimal"
Je suis sous ubuntu, je n'ai pas testé sous windows
Bonne soirée
pleco27

**Reply by pleco27** [reply]
Bonsoir, je voulais être certain de l'anomalie: test effectué ce soir d'une machine sous Linux et Windows, sous Firefox et Chrome: Anomalie identique quel que soit l'OS, quelque soit le navigateur...
Bonne soirée
pleco27

**Reply by ynaessens** [reply]
Bonjour,
je fais le même constat.
Il faudrait que Jérôme étudie la question : pour cela ouvrez un nouveau sujet, ce sera plus clair pour les autres utilisateurs. Merci

**Reply by pleco27** [reply]
Bonsoir ynaessens
ok: je crée un nouveau sujet de suite, même si le souci n'est pas bloquant
pleco27
